### PR TITLE
Implement (DataFrame|Series).plot.hist in plotly

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR implements `(DataFrame|Series).plot.hist` in plotly:

This can be tested via: https://mybinder.org/v2/gh/HyukjinKwon/koalas/plotly-histogram?filepath=docs%2Fsource%2Fgetting_started%2F10min.ipynb

Example:

```python
# Koalas
import databricks.koalas as ks
ks.options.plotting.backend = "plotly"
kdf = ks.DataFrame({
    'a c': [1, 2, 3, 4, 5, 6, 7, 8, 9, 15, 50],
    'b': [2, 3, 4, 5, 7, 9, 10, 15, 10, 20, 20]
})
(kdf + 100).plot.hist()

# pandas
import pandas as pd
pd.options.plotting.backend = "plotly"
pdf = pd.DataFrame({
    'a c': [1, 2, 3, 4, 5, 6, 7, 8, 9, 15, 50],
    'b': [2, 3, 4, 5, 7, 9, 10, 15, 10, 20, 20]
})
(pdf + 100).plot.hist()
```

Screen Shot 2021-01-12 at 10 12 47 PM

Screen Shot 2021-01-12 at 10 12 52 PM

 NOTE that the output is a bit different because:
 - We use Spark for histogram calculation and it's a bit different from pandas'
 - Histogram plot in plotly cannot be directly used in our case but should work around by leveraging bar charts because we don't use plotly to calculate histogram, see https://plotly.com/python/histograms/.

### CODE REVIEW

File: databricks/koalas/plot/core.py [hunk 1]
```diff
@@ -338,31 +344,41 @@ def _get_plot_backend(backend=None):
 
             KoalasPlotAccessor._backends["matplotlib"] = module
 
+        if backend == "plotly":
```
Comment (HyukjinKwon) [inline databricks/koalas/plot/core.py:347]: Here I avoided to monkey patch plotly. I will need to refactor codes here in another PR .. here needs some refactoring.

File: databricks/koalas/plot/matplotlib.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 from distutils.version import LooseVersion
 
-import matplotlib
+import matplotlib as mat
```
Comment (HyukjinKwon) [inline databricks/koalas/plot/matplotlib.py:19]: Here is due to the conflict when you import, `from databricks.koalas.plot import matplotlib`
As a module `databricks.koalas.plot.matplotlib` vs `matplotlib` library being imported at `databricks.koalas.plot`. I will fix it back in the next PR that refactors.

File: databricks/koalas/plot/core.py [hunk 3]
```diff
@@ -338,31 +343,42 @@ def _get_plot_backend(backend=None):
...
+            try:
+                # test if plotly can be imported
+                import plotly  # noqa: F401
+                from databricks.koalas.plot import plotly as module
+            except ImportError:
+                raise ImportError(
+                    "matplotlib is required for plotting when the "
+                    "default backend 'matplotlib' is selected."
```
Comment (ueshin) [inline databricks/koalas/plot/core.py:354]: The error message sounds weird? We should replace "matplotlib" with "plotly" here?
Also "default" is not needed?

File: databricks/koalas/plot/plotly.py [hunk 4]
```diff
@@ -15,17 +15,21 @@
...
-    return plot
+    # Koalas specific plots
+    if kind == "pie":
+        return plot_pie(data, **kwargs)
+    if kind == "hist":
+        # Note that here data is a Koalas DataFrame or Series unlike other type of plots.
+        return plot_histogram(data, **kwargs)
+
+    # Other plots.
+    return plotly.plot(data, kind, **kwargs)
```
Comment (ueshin) [inline databricks/koalas/plot/plotly.py:32]: Nice! 👍

**Review by ueshin — COMMENTED** [review]
Seems good.

File: databricks/koalas/plot/core.py (hunk 3 above)
Comment (HyukjinKwon) [reply databricks/koalas/plot/core.py:354]: oops

**Pull request merged**